kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -387,6 +397,14 @@ SetOfKeyValuePairs DBAdapter::lastReachableBlockDbUpdates(const SetOfKeyValuePai
 
   for (const auto &[k, v] : non_provable_kvs) {
     dbUpdates[DBKeyManipulator::genNonProvableBlockDbKey(blockId, k)] = v;
+    if (blockId) {
```

Comment:
Maybe it will be clearer if we compare fro greater than `INITIAL_GENESIS_BLOCK_ID` instead of 0.

Reply:
Done